engine/src/main/java/org/hibernate/search/backend/impl/DeletionQueryMapper.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.impl;
+
+import org.apache.lucene.search.Query;
+import org.hibernate.search.backend.DeletionQuery;
+import org.hibernate.search.util.impl.ScopedAnalyzer;
+
+/**
+ * @author Martin Braun
+ */
+public interface DeletionQueryMapper {
```

Comment:
It's not a problem per-se, I'm more trying to figure the rationale behind the design. What other implementations do you envision? A comment would be good, describing possible means of extension. Otherwise, an interface with a single implemenation will aways make us (at least me;)) wonder what it is about and whether the interface/implementation split is actually needed.